### PR TITLE
core: disable quic servers on mainnet-beta

### PR DESCRIPTION
**Pull request by t-nelson**
Add hidden flag to enable QUIC servers for `ClusterType::MainnetBeta` to allow for further testing before a wider roll out.

### CODE REVIEW

File: core/src/validator.rs [hunk 1]
```diff
@@ -980,6 +982,18 @@ impl Validator {
             &connection_cache,
         );
 
+        let enable_quic_servers = if genesis_config.cluster_type == ClusterType::MainnetBeta {
+            config.enable_quic_servers
+        } else {
+            if config.enable_quic_servers {
+                warn!(
+                    "ignoring --enable-quic-servers. QUIC is always enable for cluster type: {:?}",
```
Comment (mvines) [inline core/src/validator.rs:990]:
```suggestion
                    "ignoring --enable-quic-servers. QUIC is always enabled for cluster type: {:?}",
```

Comment (t-nelson) [reply core/src/validator.rs:990]: f

File: validator/src/main.rs [hunk 2]
```diff
@@ -1210,6 +1210,11 @@ pub fn main() {
                 .takes_value(false)
                 .help("Use QUIC to send transactions."),
         )
+        .arg(
+            Arg::with_name("enable_quic_servers")
+                .hidden(true)
```
Comment (mvines) [inline validator/src/main.rs:1215]: should this arg conflict with `tpu_use_quic`, or doesn't matter?

Comment (t-nelson) [reply validator/src/main.rs:1215]: don't think it matters, they're independent resources (locally)

**Review by mvines — DISMISSED** [review]
lgtm, just nits. thanks!

**Review by jstarry — APPROVED** [review]

File: core/src/validator.rs [hunk 3]
```diff
@@ -980,6 +982,18 @@ impl Validator {
             &connection_cache,
         );
 
+        let enable_quic_servers = if genesis_config.cluster_type == ClusterType::MainnetBeta {
+            config.enable_quic_servers
+        } else {
+            if config.enable_quic_servers {
```
Comment (pgarg66) [inline core/src/validator.rs:988]: Should this be a warning when enable_quic_servers is false? For 'true', this check is noop anyway.

**Review by pgarg66 — APPROVED** [review]
Minor comment, looks good otherwise.

File: core/src/validator.rs (hunk 3 above)
Comment (t-nelson) [reply core/src/validator.rs:988]: the arg takes no values, so passing it anywhere but MB is a noop.  idea was to tell the operator "you did a thing that is superfluous"

Comment (pgarg66) [reply core/src/validator.rs:988]: ah ok. makes sense.

**Pull request merged**